StripePaymentSheet/StripePaymentSheetTests/PaymentSheet/EmbeddedPaymentElementTest.swift

```diff
@@ -0,0 +1,134 @@
+        sut.update(intentConfiguration: paymentIntentConfig) { update2Result in
+            // ...should succeed.
+            XCTAssertEqual(update2Result, .succeeded)
+            XCTAssertFalse(sut.loadResult.intent.isSettingUp)
+            XCTAssertTrue(sut.loadResult.intent.isPaymentIntent)
+            // TODO: Test paymentOption updates correctly. 
+
+            // Sanity check that the analytics...
+            let analytics = STPAnalyticsClient.sharedClient._testLogHistory
+            let loadStartedEvents = analytics.filter { $0["event"] as? String == "mc_load_started" }
+            let loadSucceededEvents = analytics.filter { $0["event"] as? String == "mc_load_succeeded" }
+            // ...have the expected # of start and succeeded events...
+            XCTAssertEqual(loadStartedEvents.count, 3)
+            XCTAssertEqual(loadSucceededEvents.count, 3)
+            // ..and all have the same session id...
+            let sessionID = analytics.first?["session_id"] as? String
+            (loadStartedEvents + loadSucceededEvents).map { $0["session_id"] as? String }.forEach {
+                XCTAssertEqual($0, sessionID)
+            }
+
+            secondUpdateExpectation.fulfill()
+        }
+        await fulfillment(of: [secondUpdateExpectation])
+    }
+
+    func testUpdateFails() async throws {
+        // Given a EmbeddedPaymentElement instance...
+        let sut = try await EmbeddedPaymentElement.create(intentConfiguration: paymentIntentConfig, configuration: configuration)
+        // ...updating w/ an invalid intent config should fail...
+        var intentConfig = paymentIntentConfig
+        intentConfig.mode = .setup(currency: "Invalid currency", setupFutureUsage: .offSession)
+        let updateResult = await sut.update(intentConfiguration: intentConfig)
+        switch updateResult {
+        case .failed(error: let error):
+            print(error.nonGenericDescription)
+        default:
+            break
+        }
+
+        // Updating should succeed after failing to update...
+        let secondUpdateExpectation = expectation(description: "Second update succeeds")
+        intentConfig.mode = .setup(currency: "USD", setupFutureUsage: .offSession)
+        // ...(using the completion block based API this time)...
+        sut.update(intentConfiguration: intentConfig) { update2Result in
+            XCTAssertEqual(update2Result, .succeeded)
+            secondUpdateExpectation.fulfill()
+        }
+        await fulfillment(of: [secondUpdateExpectation])
+    }
+
+    func testUpdateCancelsInFlightUpdate() async throws {
+        // Given a EmbeddedPaymentElement instance...
+        let sut = try await EmbeddedPaymentElement.create(intentConfiguration: paymentIntentConfig, configuration: configuration)
+        // ...updating...
+        async let _updateResult = await sut.update(intentConfiguration: paymentIntentConfig)
```

Comment:
If we don't want the first update to finish, I would have expected the first call to update to not have an await?